ui/components/ui/survey-toast/survey-toast.tsx

```diff
@@ -0,0 +1,128 @@
+import React, { useEffect, useState, useContext } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import fetchWithCache from '../../../../shared/lib/fetch-with-cache';
+import { DAY } from '../../../../shared/constants/time';
+import { MetaMetricsContext } from '../../../contexts/metametrics';
+import {
+  MetaMetricsEventCategory,
+  MetaMetricsEventName,
+} from '../../../../shared/constants/metametrics';
+import {
+  getSelectedInternalAccount,
+  getLastViewedUserSurvey,
+  getUseExternalServices,
+  getParticipateInMetaMetrics,
+} from '../../../selectors';
+import { setLastViewedUserSurvey } from '../../../store/actions';
+import { Toast } from '../../multichain';
+
+type Survey = {
+  url: string;
+  description: string;
+  cta: string;
+  surveyId: number;
+};
+
+export function SurveyToast() {
+  const [survey, setSurvey] = useState<Survey | null>(null);
+  const dispatch = useDispatch();
+  const trackEvent = useContext(MetaMetricsContext);
+  const lastViewedUserSurvey = useSelector(getLastViewedUserSurvey);
+  const participateInMetaMetrics = useSelector(getParticipateInMetaMetrics);
+  const basicFunctionality = useSelector(getUseExternalServices);
+  const internalAccount = useSelector(getSelectedInternalAccount);
+
+  useEffect(() => {
+    if (!basicFunctionality) {
+      return;
+    }
+
+    const surveyId = 1;
+    const surveyUrl = `https://accounts.dev-api.cx.metamask.io/v1/users/${internalAccount.address}/surveys?surveyId=${surveyId}`;
```

Comment:
Update this so that we support a prod endpoint

Reply:
The `surveyUrl` is computed inside the `useEffect`. Since it depends on `internalAccount.address`, you could compute it outside the effect and memoize it to prevent unnecessary re-computation, improving performance although in this case it’s a minor optimization here.

```
const surveyUrl = useMemo(
  () =>
    `https://accounts.dev-api.cx.metamask.io/v1/users/${internalAccount.address}/surveys?surveyId=${surveyId}`,
  [internalAccount?.address],
);
```

Reply:
Done